exporters/metrics/ocagent/src/main/java/io/opencensus/exporter/metrics/ocagent/MetricsProtoUtils.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.metrics.ocagent;
+
+import com.google.protobuf.DoubleValue;
+import com.google.protobuf.Int64Value;
+import io.opencensus.common.Function;
+import io.opencensus.common.Functions;
+import io.opencensus.common.Timestamp;
+import io.opencensus.metrics.export.Distribution;
+import io.opencensus.metrics.export.Summary;
+import io.opencensus.proto.metrics.v1.DistributionValue;
+import io.opencensus.proto.metrics.v1.LabelKey;
+import io.opencensus.proto.metrics.v1.LabelValue;
+import io.opencensus.proto.metrics.v1.Metric;
+import io.opencensus.proto.metrics.v1.MetricDescriptor;
+import io.opencensus.proto.metrics.v1.Point;
+import io.opencensus.proto.metrics.v1.SummaryValue;
+import io.opencensus.proto.metrics.v1.TimeSeries;
+import io.opencensus.proto.resource.v1.Resource;
+import javax.annotation.Nullable;
+
+/** Utilities for converting Metrics APIs in OpenCensus Java to OpenCensus Metrics Proto. */
+final class MetricsProtoUtils {
+
+  static Metric toMetricProto(
+      io.opencensus.metrics.export.Metric metric,
+      @Nullable io.opencensus.resource.Resource resource,
+      boolean alreadySent) {
+    Metric.Builder builder = Metric.newBuilder();
+    if (alreadySent) {
```

Comment:
Based on feedback from @fabxc let's postpone this optimization. He mentioned that this may cause memory overhead on the server. Same for buckets definition.

Reply:
I see, updated to not do the optimization and left a TODO here.